sympy/logic/boolalg.py

```diff
@@ -2898,21 +2901,52 @@ def simplify_logic(expr, form=None, deep=True, force=False):
     # get variables in case not deep or after doing
     # deep simplification since they may have changed
     variables = _find_predicates(expr)
+    # Replace variables with Dummys in case there are Relationals to possibly
+    # reduce the number of variables
+    repl = dict()
+    undo = dict()
+    from sympy.core.relational import Relational
```

Comment:
Can this import and the one a few lines down get moved out of the function or at least to the beginning? I don't fully understand the motivations to where they are located and appreciate wanting to keep them near the code that needs them...but just checking.

Reply:
Not outside due to circular imports. Of course they can go to the top of the function, but I thougt I'd keep them nearby (Dummy is not needed unless in the if.)

Reply:
If there are inlined imports, I usually try to import them where needed if there are returns before they are needed. If not, at the top of the method/function.

If only needed inside an if-statement I try to import them there.

(Unless the import ends up in a loop.)

Reply:
Can this potentially be done with "expr.is_Relational"?

Reply:
You right. I'll give it a try!

Reply:
Ahh, no, it cannot. Since expr is the logic expression: 

```
And(x, y > 0).is_Relational
Out[4]: False
```

Reply:
Would something like this be suitable:
`any([expr.is_Relational for expr in preorder_traversal(And(x, y > 0))])`

Reply:
Or does that also cause circular imports?

In the end I suppose the issue really is that sympy.core heavily makes use of `Boolean`, but `Boolean` also shares the same source file as functions that use sympy.core.

Reply:
I think it may work (just quickly looking at the import patterns), but not really sure if it is worth it compared to having a more directly readable code.

Yes, parts of boolalg (Boolean and And, Or etc) should probably move to core. I made a quick attempt some time back and if I recall correctly it was still quite feasible. However, I am not sure it actually solves the circular import since relational will have to import from boolalg (or whatever that should be called) and vice versa.

Reply:
Also, there are import from relational in quite a few functions and methods in boolalg, so from that perspective it is maybe not very beneficial either.

Reply:
An option, after browsing the other code, is to use self.atoms(Relational) for the initial set. Will not help with the import issue, but simplify the code otherwise. (As we are only interested in replacing the Relationals.)

Reply:
It would be nice if `And(x, y > 0).find(lambda expr: expr.is_Relational)` had an alias like 
`And(x, y > 0).find(is_Relational = True)`